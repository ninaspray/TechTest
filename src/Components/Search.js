import React, { useState } from "react";
import getImages from '../Requests/getImages';
import PropTypes from "prop-types";

import "../Styles/Search.css";

const Search = ({ setSearchResults }) => {
   
    const [value, setValue] = useState();
    const handleSubmit = async(event) => {
    event.preventDefault();
    setSearchResults(await getImages(value));
  }
    return (   
    
        <div className="Search">
        <form className="search-form" onSubmit={handleSubmit}>
        <input className="Search-field" 
        type="text"
        onChange={(e) => setValue(e.target.value)}>
        </input>
        <button className="search-comp" type="submit">
            Search!
        </button>
        </form>
        </div>
    );
};
Search.propTypes = {
    setSearchResults: PropTypes.func.isRequired,
  };
export default Search;
