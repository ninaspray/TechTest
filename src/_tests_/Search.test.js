import React from 'react';
import { render } from "@testing-library/react";
import Search from '../Components/Search';

describe("Search", () => {
    const { asFragment } = render(<Search />);

    it("renders correctly", () => {
        expect(asFragment()).toMatchSnapshot();
    });
});