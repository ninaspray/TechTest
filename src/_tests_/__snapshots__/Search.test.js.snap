// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <div
    class="Search"
  >
    <form
      class="search-form"
    >
      <input
        class="Search-field"
        type="text"
      />
      <button
        class="search-comp"
        type="submit"
      >
        Search!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
